components/automate-gateway/api/iam/v2beta/users.proto

```diff
@@ -51,4 +51,14 @@ service Users {
     option (chef.automate.api.iam.policy).resource = "iam:users:{id}";
     option (chef.automate.api.iam.policy).action = "iam:users:update";
   };
+  rpc UpdateSelf (UpdateSelfReq) returns (UpdateSelfResp) {
+    option (google.api.http).put = "/iam/v2beta/self/{id}";
+    option (google.api.http).body = "*";
+    // The users namespace, as opposed for auth:users, is for actions users
+    // are allowed to perform on themselves by default without admin permissions.
+    option (chef.automate.api.policy).resource = "users:{id}";
+    option (chef.automate.api.policy).action = "update";
+    option (chef.automate.api.iam.policy).resource = "iam:usersSelf:{id}";
```

Comment:
But I still maintain that the extra resource and action is avoidable 😉

Reply:
I'd argue we should remove the ability for users to hit UpdateUser on themselves by default.

Reply:
Gonna merge for now so we can keep going, but feel free to refactor if you have a better approach :)